ipn/ipnlocal/local.go

```diff
@@ -4048,11 +3980,52 @@ func (b *LocalBackend) SwitchProfile(profile ipn.ProfileID) error {
 	return b.resetForProfileChangeLockedOnEntry()
 }
 
+func (b *LocalBackend) initTKALocked() error {
+	cp := b.CurrentProfile()
+	if cp.ID == "" {
+		b.tka = nil
+		return nil
+	}
+	if b.tka != nil && b.tka.profile == cp.ID {
+		// Already initialized.
+		return nil
+	}
+	root := b.TailscaleVarRoot()
+	if root == "" {
+		b.logf("network-lock unavailable; no state directory")
+		return nil
```

Comment:
do you have to `b.tka = nil` in this failure path too?

Reply:
No, it can never have been initialized if the `TailscaleVarRoot` is empty.

Reply:
but sure.